helix-view/src/document.rs

```diff
@@ -1054,14 +1063,41 @@ impl Document {
 
         self.detect_indent_and_line_ending();
 
-        match provider_registry.get_diff_base(&path) {
-            Some(diff_base) => self.set_diff_base(diff_base),
-            None => self.diff_handle = None,
+        self.reload_diff_data(&path);
+
+        Ok(())
+    }
+
+    fn reload_diff_data(&mut self, path: &Path) {
+        let diff_data = &mut self.diff_data;
+
+        diff_data.version_control_head = diff_data.source.get_current_head_name(path);
```

Comment:
Switching a file to the file diff source should not be hiding the vcs statusline element. I think this should really just be a global thing instead of being determined for each buffer seperatly . You can then also use autosetectikn of the cwd there (like the git status picker). 

It's not an absolute must to do this in this PR but if you feel up to it, I think it would be nice to include here.